### ISSUE TITLE
Research: Track instance use in Azure

### ISSUE THREAD

**Issue by infinitewarp** [cloudigrade/cloudigrade#1149]
## Summary

As a cloudigrade user in Azure, I want cloudigrade to know when I'm running instances (VMs) in Azure.

## Assumptions and Questions

- Previous investigation and potential solution: https://github.com/cloudigrade/cloudigrade/issues/839#issuecomment-910879347
- Whatever solution we implement, we probably want to go through Lighthouse since that gives us a broad "reader" role.
- Do we want to use event hub? Can we configure it to feed data from the customer to us/our tenant?
- Do we want to be very simple and just scrape "what is running now" on a cron?
  - We may want to seriously consider this (because it should be straightforward and reliable) but as a last option because it is likely to miss many things.
- Why is the Azure log not like AWS CloudTrail? With CloudTrail we are confident to read "everything", but that's because CloudTrail has an option to track only "write" events which are far fewer. Azure log, however, includes all read and write events (as of last investigation).
- For now, DO NOT BE CONCERNED ABOUT the problem of image inspection. We are knowingly ignoring that problem for now because we may not need to deal with it directly. We may be able to rely on tags or other metadata to "identify RHEL".
- Expect that if we turn this prototype script into real service code, we will likely throw away any early data as we better understand the problem space and understand how we can/cannot identify RHEL. It's okay to write "throwaway code" to prove out the possible integration.

## Expected output

- [x] a script or Jupyter lab/notebook that demonstrates some functionality for getting usage information
  - usage should include some ID for the VM that ran and when it ran (start? stop? is currently?)

## Time box

- 1 week. Check in with the rest of the team, and maybe take more time after that?

**Comment by katherine-black** [comment]
The tl;dr -- https://github.com/cloudigrade/cloudigrade/issues/839#issuecomment-910879347 remains the most straightforward way for us to do this.

More information -- 

Export activity log option in activity monitor has been expanded, and originally gave me hope in being able to leverage the lighthouse connection to a customers subscription to be able to send data back to. 

![image](https://user-images.githubusercontent.com/34514071/166509633-c7f4c5c0-439c-4df1-a167-f3eac4e1dc6d.png)

Unfortunately, that convience of an integration would require looking into another integration with microsoft. Link from pic above is here: https://docs.microsoft.com/en-us/azure/partner-solutions/overview . Another article on partner integrations with Azure Monitor: https://docs.microsoft.com/en-us/azure/azure-monitor/partners . That appears to be an ideal place for cloudigrade, but contacts need to be established first.

So why send the data to an Activity Hub or a Storage Account? Unfortunately the same restrictions as before still exist, the cloudigrade azure account is not under the same AD umbrella, which contains subs, which contain our access, which means cloudigrade can not achieve the goal of not costing the customer any money. There is no option to take data out of this hierarchy***** and stream it into a storage account in a different AD. Not without creating credentials or special access for each of our customers, and I do not expect this to be trivial, or maybe even functional.

\***** without once again incurring money on the customer account. What _could_ be done is that cloudigrade could spin up an event hub or a storage account inside the customers subscription, to which cloudigrade would then stream activity monitor logs to. This has the added advantage of keeping the customer data.. with the customer. On the other hand, storage accounts and event hubs are not cheap. Even less so when some tool is connecting to those services, and is processing data in them. 

With all that said, unfortunately while the activity monitor was expanded, it does not give the cloudigrade team any neccesarily new ready to use tools that they could leverage now, instead, the method described in [this spike](https://github.com/cloudigrade/cloudigrade/issues/839#issuecomment-910879347) remains the straightforward way to do this. If the team is willing to make some changes about how cloudigrade operates, other potential solutions open up.

**Comment by infinitewarp** [comment]
Closing this issue because the notes look good. Thanks!
